romana-install/romana-setup

```diff
@@ -80,66 +126,69 @@ if (( $# > 0 )); then
 			;;
 		*)
-			echo "Unknown action '$1' for target '$target'."
+			echo "Unknown action '$1' for platform '$platform'."
 			usage
 			exit 1
 	esac
 fi
 
-# Check Ansible is installed. May need to check additional commands
-required=( ansible-playbook )
+# Check if required commands are installed.
 for i in "${required[@]}"; do
 	if ! command -v "$i" > /dev/null; then
-		echo "Required command '$i' not found. Please check your Ansible installation."
+		echo "Required command '$i' not found. Please check your installation."
 		exit 1
 	fi
 done
 
-case "$target" in
-	aws)
+stack_data_dir="stacks/${stack_name}_${platform}_${stack_type}"
+mkdir -p "${stack_data_dir}"
+
+ansible_args=( -e stack_name="$stack_name" -e platform="$platform" -e stack_type="$stack_type" -e stack_data_dir="$stack_data_dir" )
+
+# Add an EC2 keypair to the instance if there's one defined.
+if [[ "$platform" == "aws" ]] && [[ -r ~/.aws/ec2_keypair ]]; then
+	read ec2_keypair _ < ~/.aws/ec2_keypair
+	if [[ "$ec2_keypair" ]]; then
+		ansible_args+=( -e ec2_keypair="$ec2_keypair" )
+	fi
+fi
+
+case "$platform" in
+	aws|vagrant)
 		case "$action" in
 			install)
-				ansible-playbook "$@" -i /dev/null create_aws.yml &&
-					ansible-playbook "$@" -i ec2.py config_aws.yml &&
-					[[ -f "aws-stackinfo.txt" ]] && cat aws-stackinfo.txt
-				;;
-			uninstall)
-				ansible-playbook "$@" -i /dev/null destroy_aws.yml
-				# Cleanup generated files
-				for f in devstack_lite.json aws-stackinfo.txt; do
-					[[ -f "$f" ]] && echo "Deleting $f" && rm "$f"
+				ansible-playbook "$@" "${ansible_args[@]}" -i localhost create_hosts.yml  || exit 1
+				ansible-playbook "$@" "${ansible_args[@]}" -i "${stack_data_dir}/inventory" post-create.yml &&
+				hosts_ready=0
+				for ((i=0; i < 5; i++)); do
+					echo "Attempting to connect to hosts"
+					if ansible -i "${stack_data_dir}/inventory" stack_nodes -m ping; then
+						hosts_ready=1
+						break
+					fi
+					echo "Retrying in 10 seconds."
+					sleep 10
+					
 				done
-				;;
-			*)
-				echo "Unknown action '$action' for target '$target'"
-				exit 1
-				;;
-		esac
-		;;
-	vagrant)
-		case "$action" in
-			install)
-				# echo "Not implemented yet"
-				ansible-playbook "$@" -i /dev/null create_vagrant.yml; # &&
-					ansible-playbook "$@" -i vagrant.py config_vagrant.yml &&
-					[[ -f "vagrant-stackinfo.txt" ]] && cat vagrant-stackinfo.txt
+				if (( ! hosts_ready )); then
+					echo "Unable to reach host(s). Exiting"
+					exit 1
+				fi
+				ansible-playbook "$@" "${ansible_args[@]}" -i "${stack_data_dir}/inventory" config.yml &&
+					[[ -f "${stack_data_dir}/stackinfo" ]] && cat "${stack_data_dir}/stackinfo"
 				;;
 			uninstall)
-				ansible-playbook "$@" -i /dev/null destroy_vagrant.yml
-				# Cleanup generated files
-				for f in Vagrantfile vagrant-stackinfo.txt; do
-					[[ -f "$f" ]] && echo "Deleting $f" && rm "$f"
-				done
+				ansible-playbook "$@" "${ansible_args[@]}" -i "${stack_data_dir}/inventory" pre-delete.yml
```

Comment:
pre-delete uses the inventory that's generated when hosts are created
delete doesn't, operating only on localhost.